crates/sui-core/src/execution_cache/writeback_cache.rs

```diff
@@ -2082,6 +2109,37 @@ impl TransactionCacheRead for WritebackCache {
         )
     }
 
+    fn transaction_executed_in_last_epoch(
+        &self,
+        digest: &TransactionDigest,
+        current_epoch: EpochId,
+    ) -> SuiResult<bool> {
+        let ticket = self
+            .cached
+            .transaction_executed_in_last_epoch
+            .get_ticket_for_read(digest);
+
+        if let Some(cached) = self.cached.transaction_executed_in_last_epoch.get(digest)
+            && let PointCacheItem::Some(()) = *cached.lock()
+        {
+            return Ok(true);
+        }
+
+        let was_executed = self
+            .store
+            .perpetual_tables
+            .was_transaction_executed_in_last_epoch(digest, current_epoch)?;
+
+        if was_executed {
+            self.cached
+                .transaction_executed_in_last_epoch
+                .insert(digest, Some(()), ticket)
+                .ok();
+        }
```

Comment:
we should have an else clause that inserts `None` in order to have both positive and negative caching. and then once we have that the cache hit path above needs to be able to return either true or false depending on if the cached value is Some or None